src/adapters/http/client.ts

```diff
@@ -0,0 +1,72 @@
+import Adapter from "../../lib/adapter.js"
+import GleeMessage from "../../lib/message.js"
+import got from "got"
+import { HttpAuthConfig, HttpAdapterConfig } from "../../lib/index.js"
+import http from "http"
+class HttpClientAdapter extends Adapter {
+
+  name(): string {
+    return "HTTP client"
+  }
+  async connect(): Promise<this> {
+    return this._connect()
+  }
+
+  async send(message: GleeMessage): Promise<void> {
+    return this._send(message)
+  }
+
+  async _connect(): Promise<this> {
+    this.emit("connect", {
+      name: this.name(),
+      adapter: this,
+      connection: http,
+      channel: this.channelNames,
+    })
+    return this
+  }
+  async _send(message: GleeMessage): Promise<void> {
```

Comment:
There's no need to have this function here. I'd move the whole logic to the `send` function instead.